### PR TITLE
Changes to support Unix Domain Socket

### PR DESCRIPTION
**Pull request by adityapinglesf**
Changes to support Unix Domain Socket. All API calls will check if UDS is enabled. If enabled, it will perform requests via unix socket. If not enabled, it will perform requests over regular SVC_PORT.

### CODE REVIEW

File: library/splunk_api_test.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    endpoint = module.params['url']
+    username = module.params['username']
+    password = module.params['password']
+    payload = module.params.get('payload', None)
+    headers = module.params.get('headers', None)
+    verify = module.params.get('verify', False)
+    status_code = module.params.get('status_code', None)
+    timeout = module.params.get('timeout', None)
+
+    if supports_uds():
```
Comment (adityapinglesf) [inline library/splunk_api_test.py:71]: default: 8089 port, even if UDS is enabled. need a flag to switch between UDS/HTTP

list of actions:

- build
- MUST deploy with UDS and non-UDS env
- Splunk/UF works as expected

File: library/splunk_api.py
Comment (adityapinglesf) [inline library/splunk_api.py:1]: line 9 and 10 needs to be in the constants file

Comment (adityapinglesf) [reply library/splunk_api.py:1]: UDS socket path needs to be relative -- not hardcoded for  opt as {splunk.home}

Comment (adityapinglesf) [reply library/splunk_api.py:1]: can it be configured and do we want splunk users to configure this??

File: roles/splunk_common/tasks/set_mgmt_mode.yml [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    owner: "{{ splunk.user }}"
+    group: "{{ splunk.group }}"
+  when: splunk.splunk_http_enabled|bool == false
+
+- name: Set Mgmt Mode TCP
+  ini_file:
+    dest: "{{ splunk.home }}/etc/system/local/server.conf"
+    option: "mgmtMode"
+    section: "httpServer"
+    value: "auto"
```
Comment (hendolim) [inline roles/splunk_common/tasks/set_mgmt_mode.yml:17]: This value is the same as line 6?

File: library/splunk_api.py
Comment (adityapinglesf) [reply library/splunk_api.py:1]: flag should be enabled by default (as it's latest feature)
end user should opt out irrespective of UDS capability of the system.

make it specific - uds or tcp

**Pull request merged**